ui/src/app/(index)/page.tsx

```diff
@@ -1,6 +1,75 @@
-const Home = () => {
+import Image from 'next/image';
+import BeforeLogin from '@/app/(index)/_components/BeforeLogin';
+import AfterLogin from '@/app/(index)/_components/AfterLogin';
+import {getCurrentUser} from '@/access/AuthenticationService';
+import Role from '@/access/Role';
+import LoginButton from '@/app/(index)/_components/LoginButton';
+import {getAnnouncements, getNumberOfGroupings, getNumberOfMemberships} from '@/services/GroupingsApiService';
+import Announcement from '@/app/(index)/_components/Announcement';
+
+const Home = async () => {
+    const [currentUser, numberOfGroupings, numberOfMemberships, announcements] = await Promise.all([
+        getCurrentUser(),
+        getNumberOfGroupings(),
+        getNumberOfMemberships(),
+        getAnnouncements()
+    ]);
+
+    const activeAnnouncements = () => {
+        if (announcements && announcements.announcements) {
+            return announcements.announcements
+                .filter((announcement) => announcement.state === 'Active')
+                .map((announcement) => announcement.message)
+        } else {
+            return []
+        }
+    }
```

Comment:
To make this more readable, it is a good practice to put guard clauses at the top for an early return:
```
const activeAnnouncements = () => {
    if (!announcements) { // Could use a better if statement
        return []
    }
    return announcements.announcements
            .filter((announcement) => announcement.state === 'Active')
            .map((announcement) => announcement.message)
     }
```